src/macros.jl

```diff
@@ -446,19 +446,28 @@ function constraint_macro(args, macro_name::Symbol, parsefun::Function)
         $parsecode
         $(refcall) = $constraintcall
     end
+
     # Determine the return type of addconstraint. This is needed for JuMP extensions for which this is different than ConstraintRef
     contype = :( constrainttype($m) )
+    creationcode = getloopedcode(variable, code, condition, idxvars, idxsets, contype, requestedcontainer)
+
+    if anonvar
+        # Anonymous constraint, no need to register it to a name
+        # nor to assign it to a variable in the user scope.
+        # We simply return the constraint reference
+        assignmentcode = variable
+    else
+        # We register the constraint reference to its name and
+        # we assign it to a variable of this name
+        assignmentcode = quote
+            registervar($m, $quotvarname, $variable)
```

Comment:
`registercon`?

Reply:
Wow, weird that it passes the tests

Reply:
Ah only the error message changes so it makes sense.